python/ray/experimental/state/state_manager.py

```diff
@@ -0,0 +1,191 @@
+
+    def unregister_raylet_client(self, node_id: str):
+        self._raylet_stubs.pop(node_id)
+
+    def register_agent_client(self, node_id, address: str, port: int):
+        options = (("grpc.enable_http_proxy", 0),)
+        channel = ray._private.utils.init_grpc_channel(
+            f"{address}:{port}", options=options, asynchronous=True
+        )
+        self._agent_stubs[node_id] = ReporterServiceStub(channel)
+
+    def unregister_agent_client(self, node_id: str):
+        self._agent_stubs.pop(node_id)
+
+    def get_all_registered_raylet_ids(self) -> List[str]:
+        return self._raylet_stubs.keys()
+
+    def get_all_registered_agent_ids(self) -> List[str]:
+        return self._agent_stubs.keys()
+
+    @apply_state_client_api_policy
+    async def get_all_actor_info(self, timeout: int = None) -> GetAllActorInfoReply:
+        request = GetAllActorInfoRequest()
+        reply = await self._gcs_actor_info_stub.GetAllActorInfo(
+            request, timeout=timeout
+        )
+        return reply
+
+    @apply_state_client_api_policy
+    async def get_all_placement_group_info(
+        self, timeout: int = None
+    ) -> GetAllPlacementGroupReply:
+        request = GetAllPlacementGroupRequest()
+        reply = await self._gcs_pg_info_stub.GetAllPlacementGroup(
+            request, timeout=timeout
+        )
+        return reply
+
+    @apply_state_client_api_policy
+    async def get_all_node_info(self, timeout: int = None) -> GetAllNodeInfoReply:
+        request = GetAllNodeInfoRequest()
+        reply = await self._gcs_node_info_stub.GetAllNodeInfo(request, timeout=timeout)
+        return reply
+
+    @apply_state_client_api_policy
+    async def get_all_worker_info(self, timeout: int = None) -> GetAllWorkerInfoReply:
+        request = GetAllWorkerInfoRequest()
+        reply = await self._gcs_worker_info_stub.GetAllWorkerInfo(
+            request, timeout=timeout
+        )
+        return reply
+
+    def get_all_job_info(self) -> Dict[str, JobInfo]:
+        # TODO(sang): Support timeout & make it async
+        # & apply the `apply_state_client_api_policy` decorator.
+        # NOTE: `apply_state_client_api_policy` assumes gRPC is used.
+        # Currently, `get_all_jobs()` uses the gRPC interface, which is
+        # InternalKVList to GCS.
+        try:
+            return self._job_client.get_all_jobs()
+        except Exception as e:
+            raise StateSourceNetworkException("Failed to query the job info.") from e
+
+    @apply_state_client_api_policy
+    async def get_task_info(
+        self, node_id: str, timeout: int = None
+    ) -> GetTasksInfoReply:
+        stub = self._raylet_stubs.get(node_id)
+        if not stub:
+            raise ValueError(f"Raylet for a node id, {node_id} doesn't exist.")
+
+        reply = await stub.GetTasksInfo(GetTasksInfoRequest(), timeout=timeout)
+        return reply
+
+    @apply_state_client_api_policy
```

Comment:
I am not sure if this decorator is a good idea, or I should just copy/paste the try/catch on all code block.

The intention is to have unified retry policy + exception policies.

Reply:
The decorator makes sense to me if it is expressive enough for retries/exceptions you want to handle